### PR TITLE
cmd: adopt changes in syspage and phfs into commands

### PR DESCRIPTION
**Pull request by HBuczynski**
## Description

Update commands after changes in syspage.
Changes in phfs are also introduced within this PR.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## How Has This Been Tested?

- [ ] Already covered by automatic testing.
- [ ] New test added: (add PR link here).
- [x] Tested by hand on: (imxrt1064, zynq7000).

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing linter checks and tests passed.
- [ ] My changes generate no new compilation warnings for any of the targets.

## Special treatment

- [ ] This PR needs additional PRs to work (list the PRs, preferably in merge-order).
- [ ] I will merge this PR by myself when appropriate.

**NOTE:**
Code compiles for `ARM` platforms. Changes in `ia32` need to be done to make code compile on all targets.

### CODE REVIEW

File: cmds/app.c [hunk 1]
```diff
@@ -71,76 +104,99 @@ static int cmd_loadApp(handler_t handler, size_t size, const char *imap, const c
...
+			(prog->dmaps = syspage_alloc(dmapSz * sizeof(u8))) == NULL) {
+		log_error("\nCannot allocate memory for %s", name);
+		return -ENOMEM;
 	}
 
-	if ((res = syspage_addProg(start, end, imap, dmap, cmdline, flags)) < 0) {
-		log_error("\nCan't add program to syspage");
+
+	if ((res = cmd_mapsAdd2Prog(prog->imaps, imapSz, imaps)) < 0 ||
+		(res = cmd_mapsAdd2Prog(prog->dmaps, dmapSz, dmaps)) < 0)
```
Comment (agkaminski) [inline cmds/app.c:159]: Please do double indentation ;)

**Review by agkaminski — COMMENTED** [review]
Apart from one comment LGTM

File: cmds/app.c (hunk 1 above)
Comment (HBuczynski) [reply cmds/app.c]: Done

**Review by agkaminski — APPROVED** [review]

**Review by Maxez — APPROVED** [review]
LGTM

**Pull request merged**